src/Microsoft.AspNet.Mvc.Localization/ViewLocalizer.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.Framework.Internal;
+using Microsoft.Framework.Localization;
+using Microsoft.Framework.Runtime;
+
+namespace Microsoft.AspNet.Mvc.Localization
+{
+    /// <summary>
+    /// A <see cref="HtmlLocalizer"/> that provides localized strings for views.
+    /// </summary>
+    public class ViewLocalizer : IViewLocalizer, ICanHasViewContext
+    {
+        private readonly IHtmlLocalizerFactory _localizerFactory;
+        private readonly string _applicationName;
+        private IHtmlLocalizer _localizer;
+
+        /// <summary>
+        /// Creates a new <see cref="ViewLocalizer"/>.
+        /// </summary>
+        /// <param name="localizerFactory">The <see cref="IHtmlLocalizerFactory"/>.</param>
+        /// <param name="applicationEnvironment">The <see cref="IApplicationEnvironment"/>.</param>
+        public ViewLocalizer(
+            [NotNull] IHtmlLocalizerFactory localizerFactory,
+            [NotNull] IApplicationEnvironment applicationEnvironment)
+        {
+            _applicationName = applicationEnvironment.ApplicationName;
+            _localizerFactory = localizerFactory;
+        }
+
+        /// <inheritdoc />
+        public LocalizedString this[string name] => _localizer[name];
```

Comment:
What is the bug number for this discussion? Note the issue is generally about treating incompletely-initialized `ICanHasViewContext` implementations consistently. Not throwing from a getter is something of a red herring since they will always throw something.

One or two affected methods and properties e.g. `DefaultViewComponentHelper.RenderInvoke()` will throw NREs in that case but most e.g. the `HtmlHelper.ViewData` getter throws a more-informative `InvalidOperationException`. It's not a common case but we don't document that these classes should _only_ be activated in one way.